### PR TITLE
feat: Implement table cell layout components

### PR DESCRIPTION
**Pull request by ling1726**
Implements table cell layout components from #24762 to break up the aforementioned PR

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |         Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `187.939 kB`<br />`52.05 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            | `33.359 kB`<br />`11.004 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |   `5.851 kB`<br />`1.964 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/a40ef34fa6d1122bdfdb708f79b90de9edb2827d'>a40ef34fa6d1122bdfdb708f79b90de9edb2827d</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24773/perf-test-react-components/Avatar-mount_base.html">1273</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24773/perf-test-react-components/Avatar-mount.html">1256</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24773/perf-test-react-components/Button-mount_base.html">923</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24773/perf-test-react-components/Button-mount.html">944</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24773/perf-test-react-components/FluentProvider-mount_base.html">1508</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24773/perf-test-react-components/FluentProvider-mount.html">1479</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24773/perf-test-react-components/FluentProviderWithTheme-mount_base.html">585</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24773/perf-test-react-components/FluentProviderWithTheme-mount.html">584</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24773/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">535</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24773/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">556</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24773/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">577</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24773/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">569</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24773/perf-test-react-components/MakeStyles-mount_base.html">1946</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24773/perf-test-react-components/MakeStyles-mount.html">1948</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24773/perf-test-react-components/SpinButton-mount_base.html">2378</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24773/perf-test-react-components/SpinButton-mount.html">2311</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-table/src/components/TableCellLayout/TableCellLayout.types.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+
+export type TableCellLayoutSlots = {
+  root: Slot<'div'>;
```
Comment (bsunderhus) [inline packages/react-components/react-table/src/components/TableCellLayout/TableCellLayout.types.ts:4]:
```suggestion
  root: NonNullable<Slot<'div'>>;
```

**Review by bsunderhus — APPROVED** [review]
LGTM 👍🏻

File: packages/react-components/react-table/src/index.ts [hunk 2]
```diff
@@ -97,3 +97,25 @@ export {
...
+} from './TableCellLayout';
+export type { TableCellLayoutProps, TableCellLayoutSlots, TableCellLayoutState } from './TableCellLayout';
+
+export {
+  TableCellPrimaryLayout,
+  tableCellPrimaryLayoutClassNames,
+  useTableCellPrimaryLayoutStyles_unstable,
+  useTableCellPrimaryLayout_unstable,
+  renderTableCellPrimaryLayout_unstable,
+} from './TableCellPrimaryLayout';
```
Comment (layershifter) [inline packages/react-components/react-table/src/index.ts:116]: TBH I think that we can get this in a single component, https://fluentsite.z22.web.core.windows.net/0.64.0/components/item-layout/definition

And may be it should not be connected to Table at all

**Pull request merged**